packages/front-end/pages/debugger.tsx

```diff
@@ -0,0 +1,303 @@
+          <div className="row">
+            <div className="col-md-4">
+              <SelectField
+                label="Select a Data Source"
+                labelClassName="font-weight-bold"
+                value={datasourceId ?? ""}
+                onChange={(newDatasource) => {
+                  form.setValue("datasourceId", newDatasource);
+                }}
+                options={datasources.map((d) => {
+                  const isDefaultDataSource = d.id === defaultDataSource;
+                  return {
+                    value: d.id,
+                    label: `${d.name}${
+                      d.description ? ` — ${d.description}` : ""
+                    }${isDefaultDataSource ? " (default)" : ""}`,
+                  };
+                })}
+                className="portal-overflow-ellipsis mb-0"
+              />
+            </div>
+          </div>
+
+          {datasourceId && (
+            <>
+              <hr className="mt-2 mb-4" />
+
+              <div className="row align-items-center">
+                <div className="col-sm-2">
+                  <SelectField
+                    required
+                    placeholder="Select identifier..."
+                    value={form.watch("userIdType")}
+                    onChange={(v) => form.setValue("userIdType", v)}
+                    options={(
+                      datasources.find((d) => d.id === datasourceId)?.settings
+                        ?.userIdTypes || []
+                    ).map((t) => {
+                      return {
+                        label: t.userIdType,
+                        value: t.userIdType,
+                      };
+                    })}
+                  />
+                </div>
+
+                <div className="col-md-4">
+                  <Field
+                    placeholder="Enter an ID"
+                    {...form.register("unitId", { required: true })}
+                    error={
+                      form.formState.errors.unitId && "Unit ID is required"
+                    }
+                    markRequired
+                  />
+                </div>
+
+                <div className="col-sm-2">
+                  <SelectField
+                    labelClassName="font-weight-bold"
+                    value={form.watch("lookbackDays") ?? ""}
+                    onChange={(value) => form.setValue("lookbackDays", value)}
+                    options={[
+                      { value: "7", label: "Last 7 days" },
+                      { value: "30", label: "Last 30 days" },
+                      { value: "90", label: "Last 90 days" },
+                      { value: "365", label: "Last year" },
+                    ]}
+                    sort={false}
+                    required
+                  />
+                </div>
+
+                <div className="col-sm-2">
+                  <Button onClick={onSubmit} disabled={loading}>
+                    Lookup
```

Comment:
nit: Should this just say "Query" so it matches other language and it's clearer that it's going to query? IDK I don't have strong opinions.